### PR TITLE
Support Python 2.7 and 3.4+

### PR DESCRIPTION
**Pull request by elyezer**
Add tox.ini file in order to test robottelo against Python 2.7 and
python 3.4. Also make robottelo support both Python 2.7 and 3.4+
versions.

### CODE REVIEW

**Comment by oshtaier** [comment]
Tried to review, but it seems too complex for me:) Caught general idea and tox, six libraries usage.

**Comment by sthirugn** [comment]
@elyezer do you want this merged after UI search?

**Comment by sthirugn** [comment]
@Ichimonji10 since you work on python 2 and 3 most of the time, can you please review this?

**Comment by Ichimonji10** [comment]
Yes, will do.

**Comment by elyezer** [comment]
@sthirugn I don't have any preference here. Is better to be after since I can adjust any missing python 2/3 compatibility  on the search code.

**Comment by elyezer** [comment]
TOX is used just to have an environment with multiple python versions.

In the default config it is running against python 2.7 and 3.4 and to run all tests for both version is a matter of just running `tox` on command line.

**Comment by sthirugn** [comment]
@elyezer yes lets merge this post UI search then.

File: robottelo/decorators.py [hunk 1]
```diff
@@ -255,7 +254,7 @@ def bz_bug_is_open(bug_id):
     try:
         bug = _get_bugzilla_bug(bug_id)
     except BugFetchError as err:
-        LOGGER.warning(err.message)
+        LOGGER.warning(err)
```
Comment (Ichimonji10) [inline robottelo/decorators.py]: :+1: The `message` attribute isn't available in Python 3.

File: tests/robottelo/test_cli.py [hunk 2]
```diff
@@ -1,8 +1,13 @@
-import mock
+import six
 import unittest2
 
 from robottelo.cli.base import Base
 
+if six.PY2:
+    import mock
+else:
+    from unittest import mock
```
Comment (Ichimonji10) [inline tests/robottelo/test_cli.py]: `mock` will be installed as a dependency no matter which Python version is used. Would it be simpler to just `import mock`?

File: tests/robottelo/test_helpers.py [hunk 3]
```diff
@@ -66,18 +71,23 @@ def test_cat_fail(self, ssh):
             return_value=FakeSSHResult([], 1, 'stderr'))
         with self.assertRaises(HostInfoError) as context:
             get_host_info()
+        if six.PY2:
+            message = context.exception.message
+        else:
+            message = str(context.exception)
```
Comment (Ichimonji10) [inline tests/robottelo/test_helpers.py]: Is there any advantage to accessing the `message` attribute directly? Coercing an exception to a string should be sufficient, no?

```python
>>> def foo():
...   raise Exception('some custom message')
... 
>>> try:
...   foo()
... except Exception as err:
...   error_message = str(err)
... 
>>> error_message
'some custom message'
>>> print(error_message)
some custom message
>>> type(error_message)
<type 'str'>
```

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,8 @@
+[tox]
+envlist = py27,py34
+[testenv]
+deps=
+    -rrequirements.txt
+    six
+    py27: mock
+commands=python -m unittest discover tests/robottelo
```
Comment (Ichimonji10) [inline tox.ini]: Does Python 2's unittest test runner deal gracefully with `subText` context managers and other new features backported by unittest2?

File: tests/robottelo/test_cli.py (hunk 2 above)
Comment (elyezer) [inline tests/robottelo/test_cli.py]: Good point. I thought on this already, but still thinking if we should update the setup.py to install only if is python 2.

**Comment by Ichimonji10** [comment]
This pull request looks nice. Can you paste some output from the command line showing that it works?

File: tox.ini (hunk 4 above)
Comment (elyezer) [inline tox.ini]: Yes, check tox output:

```
 tox
GLOB sdist-make: /home/elyezer/code/robottelo/setup.py
py27 inst-nodeps: /home/elyezer/code/robottelo/.tox/dist/robottelo-0.1.0.zip
py27 installed: ddt==1.0.0,ecdsa==0.13,fauxfactory==2.0.8,funcsigs==0.4,Inflector==2.0.11,linecache2==1.0.0,mock==1.3.0,nailgun==0.23.0,numpy==1.10.1,packaging==15.3,paramiko==1.15.4,pbr==1.8.1,pycrypto==2.6.1,pygal==2.0.9,python-bugzilla==1.2.2,pyxdg==0.25,requests==2.8.1,robottelo==0.1.0,selenium==2.48.0,six==1.10.0,traceback2==1.4.0,unittest2==1.1.0,wheel==0.24.0
py27 runtests: PYTHONHASHSEED='1173353588'
py27 runtests: commands[0] | python -m unittest discover tests/robottelo
..WARNING:robottelo.decorators:
.......WARNING:robottelo.decorators:
........................................
----------------------------------------------------------------------
Ran 49 tests in 0.038s

OK
py34 inst-nodeps: /home/elyezer/code/robottelo/.tox/dist/robottelo-0.1.0.zip
py34 installed: ddt==1.0.0,ecdsa==0.13,fauxfactory==2.0.8,Inflector==2.0.11,linecache2==1.0.0,nailgun==0.23.0,numpy==1.10.1,packaging==15.3,paramiko==1.15.4,pycrypto==2.6.1,pygal==2.0.9,python-bugzilla==1.2.2,pyxdg==0.25,requests==2.8.1,robottelo==0.1.0,selenium==2.48.0,six==1.10.0,traceback2==1.4.0,unittest2==1.1.0,wheel==0.24.0
py34 runtests: PYTHONHASHSEED='1173353588'
py34 runtests: commands[0] | python -m unittest discover tests/robottelo
.............WARNING:robottelo.decorators:
.......WARNING:robottelo.decorators:
.............................
----------------------------------------------------------------------
Ran 49 tests in 0.050s

OK
__________________________________________________________________________________ summary __________________________________________________________________________________
  py27: commands succeeded
  py34: commands succeeded
  congratulations :)
```

File: tests/robottelo/test_helpers.py (hunk 3 above)
Comment (elyezer) [inline tests/robottelo/test_helpers.py]: Absolutely. Will update. Thanks for this comment.

File: tests/robottelo/test_cli.py (hunk 2 above)
Comment (Ichimonji10) [inline tests/robottelo/test_cli.py]: Conditional dependencies cannot be added in `setup.py`. Instead, they must be added in a requirements file. See [here](https://github.com/SatelliteQE/nailgun/blob/master/requirements-dev.txt#L3) for an example.

File: tox.ini (hunk 4 above)
Comment (Ichimonji10) [inline tox.ini]: :+1:

File: tests/robottelo/test_cli.py (hunk 2 above)
Comment (elyezer) [inline tests/robottelo/test_cli.py]: :+1:

**Comment by elyezer** [comment]
Rebased after the new search merge and migrated more imports. Now `py.test --collect-only` works ok on both python 2 and 3.

**Comment by sthirugn** [comment]
ACK

**Comment by omaciel** [comment]
ACK

**Pull request merged**